django/core/management/commands/shell.py

```diff
@@ -69,9 +69,15 @@ def python(self, options):
                     continue
                 if not os.path.isfile(pythonrc):
                     continue
-                with suppress(NameError):
-                    with open(pythonrc) as handle:
-                        exec(compile(handle.read(), pythonrc, 'exec'), imported_objects)
+                with open(pythonrc) as handle:
+                    pythonrc_code = handle.read()
+
+                try:
+                    exec(compile(pythonrc_code, pythonrc, 'exec'), imported_objects)
+                except:
```

Comment:
May be he wanted to catch system exiting exceptions  - BaseException, SystemExit, KeyboardInterrupt and GeneratorExit. But i fail to see the reason for this except `KeyboardInterrupt `